src/pykiso/test_result/multi_result.py

```diff
@@ -99,9 +116,46 @@ def startTest(self, test: Union[BasicTest, BaseTestSuite]) -> None:
 
         :param test: running testcase
         """
+        self.handle_log_file_strategy(test)
+
         for result in self.result_classes:
             result.startTest(test)
 
+    def handle_log_file_strategy(self, test: Union[BasicTest, BaseTestSuite]) -> None:
+        """Handle the log file strategy for the given test.
+
+        :param test: The test case or test suite being executed.
+        """
+        if self._log_file_strategy is None:
+            return
+
+        log_options = get_logging_options()
+
+        if test.__class__ in self._list_test_results and self._log_file_strategy == "testCase":
+            log_path = self.current_log_file
+        else:
+            log_file_name = util.strclass(test.__class__).replace(".", "_").replace("-", "_")
+            if self._log_file_strategy == "testRun":
+                log_file_name += "_" + test._testMethodName
+
+            log_file_name += f"_{datetime.today().strftime('%Y%d%m%H%M%S')}.log"
+            log_path = log_options.log_path.parent / log_file_name
+
+        # Setup the file handler for logging
+        log_format = logging.Formatter("%(asctime)s [%(levelname)s] %(module)s:%(lineno)d: %(message)s")
+        root_logger = logging.getLogger()
+        file_handler = logging.FileHandler(log_path, "a+")
+        file_handler.setFormatter(log_format)
+        file_handler.name = "strategy_log_file_handler"
+        # always include internal logs in log files
+        file_handler.setLevel(get_internal_level(log_options.log_level))
+        root_logger.addHandler(file_handler)
```

Comment:
Why do you need to set again a file-handler?

Reply:
It was requested by @Pog3k that we keep the original log file that holds all the test log (so the default file handler), and we create a new file when the strategy is active so this file handler.